<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LevelWindow</class>
 <widget class="QMainWindow" name="LevelWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>996</width>
    <height>611</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <property name="margin">
     <number>2</number>
    </property>
    <item row="1" column="0">
     <widget class="MyGraphicsView" name="graphicsView">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>480</width>
        <height>320</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>160000</width>
        <height>160000</height>
       </size>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QScrollBar" name="verticalScrollBar">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QScrollBar" name="horizontalScrollBar">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QToolButton" name="toolButton_updateRect">
      <property name="text">
       <string>[]</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QFrame" name="frame">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="margin">
        <number>5</number>
       </property>
       <item>
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>X</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinBox_frameX"/>
       </item>
       <item>
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Y</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinBox_frameY"/>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>996</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuLevel">
    <property name="title">
     <string>Level</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_as"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuMoon">
    <property name="title">
     <string>Editor</string>
    </property>
    <addaction name="actionAnimation"/>
    <addaction name="actionLevel"/>
   </widget>
   <addaction name="menuMoon"/>
   <addaction name="menuLevel"/>
   <addaction name="menuEdit"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="dockWidget">
   <property name="windowTitle">
    <string>Elements</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QComboBox" name="comboBox_elementType">
       <item>
        <property name="text">
         <string>select element type</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Image</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Polygon</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="widget" native="true">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="margin">
         <number>0</number>
        </property>
        <item>
         <widget class="QPushButton" name="pushButton_addElement">
          <property name="text">
           <string>+</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_delElement">
          <property name="text">
           <string>-</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QListWidget" name="listWidget_elements"/>
     </item>
     <item>
      <widget class="QWidget" name="widget_2" native="true">
       <layout class="QGridLayout" name="gridLayout">
        <property name="margin">
         <number>4</number>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>X</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Y</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="spinBox_y">
          <property name="minimum">
           <number>-1000</number>
          </property>
          <property name="maximum">
           <number>2000</number>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QDoubleSpinBox" name="doubleSpinBox_z">
          <property name="maximum">
           <double>10.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>1.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Z</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Scale</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QDoubleSpinBox" name="doubleSpinBox_scale">
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="minimum">
           <double>0.100000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>1.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="spinBox_x">
          <property name="minimum">
           <number>-1000</number>
          </property>
          <property name="maximum">
           <number>2000</number>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Physic</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QComboBox" name="comboBox_physElement">
          <item>
           <property name="text">
            <string>None</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Static</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Dynamic</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QFrame" name="frame_polygon">
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Point</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="spinBox_polygonPoint"/>
        </item>
        <item row="0" column="2">
         <widget class="QPushButton" name="pushButton_polygonEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Edit</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QPushButton" name="pushButton_polygonAdd">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>+</string>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QPushButton" name="pushButton_polygonDel">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>-</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionNew">
   <property name="text">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_as">
   <property name="text">
    <string>Save as</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionAnimation">
   <property name="text">
    <string>Animation</string>
   </property>
   <property name="shortcut">
    <string>Meta+Ctrl+A</string>
   </property>
  </action>
  <action name="actionLevel">
   <property name="text">
    <string>Level</string>
   </property>
   <property name="shortcut">
    <string>Meta+Ctrl+L</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header location="global">mygraphicsview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>comboBox_elementType</sender>
   <signal>activated(int)</signal>
   <receiver>LevelWindow</receiver>
   <slot>setElementType(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>113</x>
     <y>70</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_addElement</sender>
   <signal>clicked()</signal>
   <receiver>LevelWindow</receiver>
   <slot>addElement()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>38</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_delElement</sender>
   <signal>clicked()</signal>
   <receiver>LevelWindow</receiver>
   <slot>delElement()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>91</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>listWidget_elements</sender>
   <signal>itemSelectionChanged()</signal>
   <receiver>LevelWindow</receiver>
   <slot>selectElement()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>113</x>
     <y>257</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_x</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LevelWindow</receiver>
   <slot>setX(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>418</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_y</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LevelWindow</receiver>
   <slot>setY(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>453</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox_z</sender>
   <signal>valueChanged(double)</signal>
   <receiver>LevelWindow</receiver>
   <slot>setZ(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>488</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox_scale</sender>
   <signal>valueChanged(double)</signal>
   <receiver>LevelWindow</receiver>
   <slot>setScale(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>523</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_physElement</sender>
   <signal>activated(int)</signal>
   <receiver>LevelWindow</receiver>
   <slot>setPhysics(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>556</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew</sender>
   <signal>triggered()</signal>
   <receiver>LevelWindow</receiver>
   <slot>newLevel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>LevelWindow</receiver>
   <slot>openLevel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>LevelWindow</receiver>
   <slot>saveLevel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_as</sender>
   <signal>triggered()</signal>
   <receiver>LevelWindow</receiver>
   <slot>saveLevelAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>verticalScrollBar</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>LevelWindow</receiver>
   <slot>scrollWindowY(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>977</x>
     <y>288</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalScrollBar</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>LevelWindow</receiver>
   <slot>scrollWindowX(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>620</x>
     <y>576</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>verticalScrollBar</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>spinBox_frameY</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>977</x>
     <y>392</y>
    </hint>
    <hint type="destinationlabel">
     <x>418</x>
     <y>42</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalScrollBar</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>spinBox_frameX</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>618</x>
     <y>576</y>
    </hint>
    <hint type="destinationlabel">
     <x>334</x>
     <y>42</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_frameX</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LevelWindow</receiver>
   <slot>scrollWindowX(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>334</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_frameY</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LevelWindow</receiver>
   <slot>scrollWindowY(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>418</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButton_updateRect</sender>
   <signal>clicked()</signal>
   <receiver>LevelWindow</receiver>
   <slot>updateFrame()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>981</x>
     <y>576</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_polygonPoint</sender>
   <signal>valueChanged(int)</signal>
   <receiver>LevelWindow</receiver>
   <slot>setPolygonPoint(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>147</x>
     <y>552</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_polygonEdit</sender>
   <signal>clicked(bool)</signal>
   <receiver>LevelWindow</receiver>
   <slot>editPolygon(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>226</x>
     <y>553</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_polygonAdd</sender>
   <signal>clicked()</signal>
   <receiver>LevelWindow</receiver>
   <slot>addPolygonPoint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>288</x>
     <y>553</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_polygonDel</sender>
   <signal>clicked()</signal>
   <receiver>LevelWindow</receiver>
   <slot>delPolygonPoint()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>341</x>
     <y>553</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLevel</sender>
   <signal>triggered()</signal>
   <receiver>LevelWindow</receiver>
   <slot>raise()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>497</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>raiseLevel()</signal>
  <signal>raiseAnimation()</signal>
  <slot>setElementType(int)</slot>
  <slot>addElement()</slot>
  <slot>delElement()</slot>
  <slot>selectElement()</slot>
  <slot>setX(int)</slot>
  <slot>setY(int)</slot>
  <slot>setZ(double)</slot>
  <slot>setScale(double)</slot>
  <slot>setPhysics(int)</slot>
  <slot>newLevel()</slot>
  <slot>openLevel()</slot>
  <slot>saveLevel()</slot>
  <slot>saveLevelAs()</slot>
  <slot>scrollWindowX(int)</slot>
  <slot>scrollWindowY(int)</slot>
  <slot>updateFrame()</slot>
  <slot>setPolygonPoint(int)</slot>
  <slot>editPolygon(bool)</slot>
  <slot>addPolygonPoint()</slot>
  <slot>delPolygonPoint()</slot>
 </slots>
</ui>
